generator/integration_tests/golden/tests/golden_thing_admin_connection_test.cc

```diff
@@ -1223,12 +1223,13 @@ TEST(GoldenThingAdminConnectionTest, AsyncGetDatabaseTooManyFailures) {
 }
 
 TEST(GoldenThingAdminConnectionTest, AsyncGetDatabaseCancel) {
-  auto cancelled = false;
+  promise<bool> cancelled;
   promise<StatusOr<google::test::admin::database::v1::Database>> p(
-      [&p, &cancelled] {
-        cancelled = true;
-        p.set_value(Status(StatusCode::kDeadlineExceeded, "try again"));
-      });
+      [&cancelled] { cancelled.set_value(true); });
+  auto cancel_completed = cancelled.get_future().then([&](future<bool> f) {
```

Comment:
Only capture `p` here, and in the `AsyncGetDatabase()` expectation?

Reply:
Done